custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -1,24 +1,114 @@
-name:  YYM-805SW Aroma diffuser
+# DPS sample: {'updated_at': 1674433099.9169242, '1': True, '2': 'large', '3': '1', '5': True, '6': 'colourful1', '8': '001fff00e8ffff'}
+name: YYM-805SW Aroma diffuser
 products:
-    - id: c0nh3LmEk0NDebrq
+  - id: c0nh3LmEk0NDebrq
 primary_entity:
   entity: fan
   dps:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
         - dps_val: small
-          value: 50
+value: 50
 secondary_entities:
   - entity: light
     name: Nightlight
+    icon: "mdi:light-recessed"
     dps:
       - id: 5
         name: switch
         type: boolean
+      - id: 8
+        name: rgbhsv
+        type: hex
+        format:
+          - name: r
+            bytes: 1
+          - name: g
+            bytes: 1
+          - name: b
+            bytes: 1
+          - name: h
+            bytes: 2
+            range:
+              min: 0
+              max: 360
+          - name: s
+            bytes: 1
+            range:
+              min: 0
+              max: 255
+          - name: v
+            bytes: 1
+            range:
+              min: 0
+              max: 255
+  - entity: select
```

Comment:
I think this should be "color_mode" on the light entity instead of a separate select.